### DISCUSSION TITLE
Add cloud storage backup.

### DISCUSSION THREAD

**Discussion by birdman3131** [gameserverapp/Platform#482] [Suggestions]
It would be great if we could have an option to add say onedrive, drop box or Google drive.

Not as a way to do transfers but just as a backup option. As far as I understand, you should be able to use the API to back up to a GSA specific folder. At least that's usually the way you can do it on a mobile app so I assume you should be able to on a desktop. That way you're not actually having access to anyone's files.

The GSA backup is really nice but does not protect from a file system corruption.

**Comment by birdman3131** [comment]
https://learn.microsoft.com/en-us/onedrive/developer/rest-api/concepts/special-folders-appfolder?view=odsp-graph-online This being a starting point for onedrive. Not sure on the others.

**Comment by ssiwm** [comment]
it would be great to be able to export the save to a cloud drive. Plus, you can recreate your server with one click if you lose your original files :)

**Comment by dagbs** [comment]
My recommendation is to add S3 compatible object storage.
This way we could either use or host our own storage server for backups.
- https://min.io/
- https://www.digitalocean.com/products/spaces
- https://us.ovhcloud.com/public-cloud/object-storage/


There are also many different open-source clients to choose from.
- https://github.com/topics/s3-client
